src/bin/client-demo.rs

```diff
@@ -148,7 +154,13 @@ fn main() {
         let requests_port = events_addr.port();
         events_addr.set_port(requests_port + 1);
         let events_socket = UdpSocket::bind(&events_addr).unwrap();
```

Comment:
This doesn't seem right. You want to `bind` to the same address you're going to `send_to`?